TestStoreRangeMergeTwoEmptyRanges: (*raftLog).appliedTo: applied(13) is out of range [prevApplied(10, committed(10))]

The following test appears to have failed:

[#9814](https://circleci.com/gh/cockroachdb/cockroach/9814):

```
I1203 04:05:42.234872 1067 client/db.go:483  failed batch: failed to push "storage/replica_command.go:1485 (*Replica).AdminMerge" id=03cf84c6 key=/Local/Range/RangeDescriptor/"" rw=false pri=85.10612849 iso=SERIALIZABLE stat=PENDING epo=0 ts=0.000000000,42 orig=0.000000000,42 max=0.000000000,42
I1203 04:05:42.234984 1067 storage/store.go:1460  on RangeLookup: txn id= key=/Min rw=false pri=85.79738112 iso=SERIALIZABLE stat=PENDING epo=1 ts=0.000000000,43 orig=0.000000000,43 max=0.000000000,0 failed to push "storage/replica_command.go:1485 (*Replica).AdminMerge" id=03cf84c6 key=/Local/Range/RangeDescriptor/"" rw=false pri=85.10612849 iso=SERIALIZABLE stat=PENDING epo=0 ts=0.000000000,42 orig=0.000000000,42 max=0.000000000,42
W1203 04:05:42.235045 1067 storage/replica.go:1401  failed to push during intent resolution: conflicting intents on [/System/Meta2/"\xff\xff"]: resolved? false
I1203 04:05:42.236235 1067 storage/replica_command.go:451  auto-gc'ed 03cf84c6 (8 intents)
E1203 04:05:42.237149 1067 raft/log.go:195  [group 2] applied(13) is out of range [prevApplied(10), committed(10)]
panic: [group 2] applied(13) is out of range [prevApplied(10), committed(10)]

goroutine 2911 [running]:
github.com/cockroachdb/cockroach/multiraft.(*raftLogger).Panicf(0xc8201bef00, 0x15ae840, 0x3c, 0xc820245a70, 0x3, 0x3)
	/go/src/github.com/cockroachdb/cockroach/multiraft/raft.go:121 +0x1a2
github.com/coreos/etcd/raft.(*raftLog).appliedTo(0xc8201eb420, 0xd)
	/go/src/github.com/coreos/etcd/raft/log.go:195 +0x1f7
github.com/coreos/etcd/raft.newRaft(0xc8201eb3b0, 0xc82028d200)
	/go/src/github.com/coreos/etcd/raft/raft.go:208 +0x924
github.com/coreos/etcd/raft.(*multiNode).run(0xc82028cde0)
	/go/src/github.com/coreos/etcd/raft/multinode.go:203 +0x423
--
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:593 +0x3bf
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc82028c180, 0xc8202daa60)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x62
FAIL	github.com/cockroachdb/cockroach/storage	1.781s
=== RUN   TestBatchBasics
I1203 04:05:31.472498 1038 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I1203 04:05:31.473483 1038 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN   TestBatchMerge
I1203 04:05:31.475604 1038 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/replica_command.go
+++ b/storage/replica_command.go
@@ -1424,7 +1424,22 @@ func (r *Replica) splitTrigger(batch engine.Engine, split *roachpb.SplitTrigger)
 		// To avoid leaving the new range unavailable as it waits to elect
 		// its leader, one (and only one) of the nodes should start an
 		// election as soon as the split is processed.
-		if r.store.StoreID() == split.InitialLeaderStoreID {
+		//
+		// If there is only one replica, raft instantly makes it the leader.
+		// Otherwise, we must trigger a campaign here.
+		//
+		// TODO(bdarnell): The asynchronous campaign can cause a problem
+		// with merges, by recreating a replica that should have been
+		// destroyed. This will probably be addressed as a part of the
+		// general work to be done for merges
+		// (https://github.com/cockroachdb/cockroach/issues/2433), but for
+		// now we're safe because we only perform the asynchronous
+		// campaign when there are multiple replicas, and we only perform
+		// merges in testing scenarios with a single replica.
+		// Note that in multi-node scenarios the async campaign is safe
+		// because it has exactly the same behavior as an incoming message
+		// from another node; the problem here is only with merges.
+		if len(split.NewDesc.Replicas) > 1 && r.store.StoreID() == split.InitialLeaderStoreID {
 			// Schedule the campaign a short time in the future. As
 			// followers process the split, they destroy and recreate their
 			// raft groups, which can cause messages to be dropped. In

